ttnn/cpp/ttnn/distributed/distributed_tensor.hpp

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/distributed/types.hpp"
+
+namespace ttnn::distributed {
+
+// Mapper interface that distributes a host tensor onto a multi-device configuration.
+class TensorToMesh {
```

Comment:
Would love to hop on call to try finding better names.
Something feels very off about current names of classes and methods.

Reply:
Ok